classes/Builder/Payload/OrderPayloadBuilder.php

```diff
@@ -0,0 +1,299 @@
+* If you did not receive a copy of the license and are unable to
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author    PrestaShop SA <contact@prestashop.com>
+*  @copyright 2007-2019 PrestaShop SA
+*  @license   http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+namespace PrestaShop\Module\PrestashopCheckout\Builder\Payload;
+
+use PrestaShop\Module\PrestashopCheckout\PaypalCountryCodeMatrice;
+use PrestaShop\Module\PrestashopCheckout\Repository\PaypalAccountRepository;
+
+/**
+ * Build the payload for creating paypal order
+ */
+class OrderPayloadBuilder extends Builder implements PayloadBuilderInterface
+{
+    /**
+     * @var array
+     */
+    private $cart;
+
+    public function __construct(array $cart)
+    {
+        $this->cart = $cart;
+
+        parent::__construct();
+    }
+
+    /**
+     * Build payload with cart details
+     */
+    public function buildFullPayload()
+    {
+        parent::buildFullPayload();
+
+        $this->buildBaseNode();
+        $this->buildAmountBreakdownNode();
+    }
+
+    /**
+     * Build payload without cart details
+     */
+    public function buildMinimalPayload()
+    {
+        parent::buildMinimalPayload();
+
+        $this->buildBaseNode();
+    }
+
+    /**
+     * Build the basic payload
+     */
+    public function buildBaseNode()
+    {
+        $countryCodeMatrice = new PaypalCountryCodeMatrice();
+        $shippingCountryIsoCode = $this->getCountryIsoCodeById($this->cart['addresses']['shipping']->id_country);
+        $payerCountryIsoCode = $this->getCountryIsoCodeById($this->cart['addresses']['invoice']->id_country);
+
+        $node = [
+            'intent' => \Configuration::get('PS_CHECKOUT_INTENT'), // capture or authorize
+            'custom_id' => (string) $this->cart['cart']['id'], // id_cart or id_order // link between paypal order and prestashop order
+            'invoice_id' => '',
+            'description' => $this->truncate('Checking out with your cart from ' . \Configuration::get('PS_SHOP_NAME'), 127),
+            'soft_descriptor' => $this->truncate(\Configuration::get('PS_SHOP_NAME'), 22),
+            'amount' => [
+                'currency_code' => $this->cart['currency']['iso_code'],
+                'value' => $this->cart['cart']['totals']['total_including_tax']['amount'],
+            ],
+            'shipping' => [
+                'name' => [
+                    'full_name' => 'Mr ' . $this->cart['addresses']['shipping']->lastname . ' ' . $this->cart['addresses']['shipping']->firstname,
```

Comment:
Hardcoded `Mr`?

Reply:
Instantiate the gender form the customer and get the gender name.